File: JE/1_22/boj_10845.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class boj_10845 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Queue<String> q = new LinkedList<>();
        int t = Integer.parseInt(br.readLine());
        String last = "";
        while (t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String input1 = st.nextToken();

            switch (input1) {
                case "push":
                    String x = st.nextToken();
                    q.add(x);
                    last = x;
                    break;
                case "pop":
                    if (q.isEmpty())
                        sb.append("-1").append("\n");
                    else
                        sb.append(q.poll()).append("\n");
                    break;
                case "size":
                    sb.append(q.size()).append("\n");
                    break;
                case "empty":
                    if (q.isEmpty())
                        sb.append("1").append("\n");
                    else
                        sb.append("0").append("\n");
                    break;
                case "front":
                    if (q.isEmpty())
                        sb.append("-1").append("\n");
                    else
                        sb.append(q.peek()).append("\n");
                    break;
                case "back":
                    if (q.isEmpty())
                        sb.append("-1").append("\n");
                    else
                        sb.append(last).append("\n");
                    break;
            }
        }
        System.out.println(sb);

    }
}

File: Lim-YongKwan/1월 28일/BOJ_10845.cpp
#include<iostream>
#include<queue>
#include<string>
#define endl "\n"
using namespace std;

int main() {

	int n;
	cin >> n;

	queue<int> silver;
	while (n--) {


		string s;
		cin >> s;
		int temp;

		if (s == "push") {
			cin >> temp;
			silver.push(temp);
		}
		else if (s == "front") {
			if (silver.size() > 0) {
				cout << silver.front() << endl;
			}
			else {
				cout << -1 << endl;
			}
		}
		else if (s == "pop") {
			if (silver.size() > 0) {
				cout << silver.front() << endl;
				silver.pop();
			}
			else {
				cout << "-1" << endl;
			}
		}
		else if (s == "size") {
			cout << silver.size() << endl;
		}
		else if (s == "empty") {
			if (silver.empty()) {
				cout << 1 << endl;
			}
			else {
				cout << 0 << endl;
			}
		}
		else if (s == "back") {
			if (silver.size() > 0) {
				cout << silver.back() << endl;
			}
			else {
				cout << -1 << endl;
			}
		}
		
	}
	return 0;
}